src/main.rs

```diff
@@ -1311,78 +1311,105 @@ fn parse_manpage_at_path(
-//
-//         # Maybe we want to skip this item
-//         skip = False
-//         if not WRITE_TO_STDOUT:
-//             # Compute the path that we would write to
-//             output_path = os.path.join(output_directory, output_file_name)
-//
-//         # Now skip if requested
-//         if skip:
-//             continue
-//
-//         try:
-//             if parse_manpage_at_path(manpage_path, output_directory):
-//                 successful_count += 1
-//         except IOError:
-//             diagnostic_indent = 0
-//             add_diagnostic('Cannot open ' + manpage_path)
-//         except (KeyboardInterrupt, SystemExit):
-//             raise
-//         except:
-//             add_diagnostic('Error parsing %s: %s' % (manpage_path, sys.exc_info()[0]), BRIEF_VERBOSE)
-//             flush_diagnostics(sys.stderr)
-//             traceback.print_exc(file=sys.stderr)
-//         flush_diagnostics(sys.stderr)
-//     print("") #Newline after loop
-//     add_diagnostic("Successfully parsed %d / %d pages" % (successful_count, total_count), BRIEF_VERBOSE)
-//     flush_diagnostics(sys.stderr)
-
-#[derive(Copy, Clone, Debug)]
-struct Progress(pub bool);
-
-// TODO Arg/output types?
+/// Get the number of digits in num
+fn num_digits(n: usize) -> usize {
+    (1.max(n) as f32).log10() as usize + 1
+}
+
+#[test]
+fn test_num_digits() {
+    assert_eq!(num_digits(1000), 4);
+    assert_eq!(num_digits(100), 3);
+    assert_eq!(num_digits(33), 2);
+    assert_eq!(num_digits(123456789012345), 15);
+    assert_eq!(num_digits(0), 1);
+}
+
+// Output: Result<cmd_name, error>
+// TODO: Result<cmd_name, CompletionsError>
 fn parse_and_output_man_pages(
-    _paths: impl Iterator<Item = PathBuf>,
-    _output_directory: PathBuf,
-    Progress(_show_progress): Progress,
+    paths: &mut [PathBuf],
+    output_directory: PathBuf,
+    show_progress: bool,
+    deroff_only: bool,
+    write_to_stdout: bool,
 ) {
-    unimplemented!();
+    let mut paths = paths;
+    paths.sort();
+
+    let total = paths.len();
+
+    let mut successful_count = 0;
+    let padding = num_digits(total);
+
+    let mut last_len: usize = 0;
+
+    if show_progress && !write_to_stdout {
+        println!(
+            "Parsing man pages and writing completions to {:?}",
+            output_directory
+        );
+    }
+
+    // As far as I can tell, we're fully supporting xz / lzma, so the checks here aren't needed
+
+    for (index, manpage_path) in paths.iter().enumerate() {
+        // foo/bar/gcc.1.gz -> gcc.1.gz
+        // TODO: Expect
+        let man_file_name = manpage_path
+            .file_name()
+            .map(|fname| fname.to_string_lossy())
+            .unwrap_or_else(|| panic!("Failed to get manfile name from {:?}", manpage_path));
+
+        // gcc.1.gz -> gcc
+        let cmd_name = man_file_name
+            .split('.')
+            .next()
+            .unwrap_or_else(|| panic!("Failed to get command name from {}", man_file_name));
```

Comment:
No need to have special panic message since this is an infallible `None`. Okay to just `unwrap` since `string.split('.').next().is_some()` is `true` for all possible strings. Maybe add a comment to that effect, e.g.:
```suggestion
            .unwrap(); // `str::split` iterator ALWAYS has a first element by definition
```

Reply:
Wasn't aware of that, thanks